---
title: "DAUR2_les6"
author: "Stephanie de Wit"
date: "7-4-2022"
output: html_document
---

```{r setup, include=FALSE}
library(png)
library(grid)
library(gridExtra)
```

__Introduction__

 Isolating DNA from metagenomics samples can be particularly difficult: 1) We do not have prior knowledge of the biodiversity in such samples, making it nearly impossible to culture such samples to meet the input DNA requirements for sequencing. 2) PCR amplification methods are discouraged, since those methods introduce a bias towards shorter DNA fragments that might skew the abundance distribution of the population. 
Hence we need to be able to obtain enough material from natural environments.

 For the species identification, we do not need to assemble full genomes. Instead, we can identify particular genomic regions that are species specific using a __k-mer analysis__. K-mer analysis is the investigation of short subsequences of length ‘k’. From every read we generate k-mers: a collection of substring per read of length ‘k’. The first k-mer starts at position 1 and ends at position ‘k’, the second k-mer starts at position 2 and ends at ‘k’+1, the third k-mer starts at position 3 and ends at position ‘k’+2 etc. __Dus indien k = 10: 1...10 + 2...11 + 3...12 etc.__
 
 Increasing the read length facilitates the simplification of assembly since you need less long reads to assemble a whole genome. Therefore the scientific community has been focusing on increasing the read length using technologies such as Pacific PacBio or Oxford Nanopore Technology. Those technologies provide reads with average lengths around __10,000 bp__, compared to 150 bp generated by Illumina sequencing. However, that length comes at the cost of __lower read accuracy__, which is 90%, compared to >99,9% from Illumina data.
 
Research question: __"Can we identify the genomes that are present in a fake dataset to confirm the species abundance using our identification algorithm?"__

_Intermezzo_

Read length van de reads in FASTQ files bestanden achterhalen: __zcat *.fastq.gz | head -n 10__. Bijv. "@A00597: __141__". Er zijn R1 en R2 FASTQ files voor een paired-end sequencing.

__Installing required software__

Use a __file.yml__ with specifications to create a virtual environment with __conda__. Create the file with the __touch__ command.

```{bash, engine.opts='-i', echo=TRUE, eval=TRUE, results="hide"}
touch setup_meta_env.yml
```

Into this file you copy paste the following:

__name: meta__

__dependencies:__

__- python=3.8__

```{bash, engine.opts='-i', echo=TRUE, eval=FALSE}
echo “y” | conda update -n base conda #Omdat we tijdens de executie van het update command geen code kunnen toevoegen om de proceed question met Y te beantwoorden
```

```{bash, engine.opts='-i', echo=TRUE, eval=FALSE}
conda env list #Alle virtual environments bekijken om duplicatie te voorkomen
conda env create --file ~/daur2/metagenomics/setup_meta_env.yml
```

__Quality control FASTQ files__

_Waarom meta virtual environment steeds deactiveren?_
```{bash, engine.opts='-i', echo=TRUE, eval=FALSE}
conda activate meta

echo "y" | conda install -c bioconda fastqc
```

```{bash, engine.opts='-i', echo=TRUE, eval=FALSE}
fastqc -o ~/daur2/metagenomics/fastqc /home/daur2/metagenomics/reader_data/HU1_MOCK1_L001_R1_001.fastq.gz

fastqc -o ~/daur2/metagenomics/fastqc /home/daur2/metagenomics/reader_data/HU1_MOCK1_L001_R2_001.fastq.gz

conda deactivate meta
```

__Exercise 6.4__

_Voor de R1 en R2 reads:_

__a)__ Total sequences = aantal reads in dataset = 40237621. _Voor elke forward een reversed_.
__b)__ De gemiddelde kwaliteit van de reads is goed: alle basen hebben een zeer hoge Phred score met een gemiddelde van 35,5. _De laatste 5 basen van R2 zijn van mindere kwaliteit, maar de gemiddelde kwaliteit = blauwe lijn ligt nog steeds boven een Phred score van 30_.
__c)__ Sequences flagged as poor quality = 0: Er zijn geen reads met een slechte kwaliteit.

__Exercise 6.5__

Gebaseerd op de indicatie dat alle basen een quality hebben van >30, kunnen we zeggen dat onze reads een accuracy van >99,9% hebben.

__Gebasseerd op deze quality control hoeven er geen reads of basen verwijderd te worden.__

_Om screenshots van de quality control toe te kunnen voegen alsof het grafieken zijn, moeten we de afbeeldingen opslaan als objects en tegen elkaar uitzetten in een arranged grid met twee kolommen._

```{r, echo=FALSE, fig.height=3, out.width= '100%', fig.cap="Fig. 1: Quality distribution: Forward read quality scores range from 0 to 40 on the Phred scale. All bases for all reads have very high quality (>30), indicating that our reads have an accuracy of >99,9%." }
img1<-rasterGrob(as.raster(readPNG("meta_lesson6_R1.png")))
img2<-rasterGrob(as.raster(readPNG("meta_lesson6_R2.png")))
grid.arrange(img1,img2,ncol=2,top=textGrob("Per base quality distribution of forward (right) and reverse (left) reads", gp=gpar(fontsize=10,font=8)))
```